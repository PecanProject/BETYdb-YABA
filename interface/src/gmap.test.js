import React from "react"
import { shallow } from 'enzyme'
import GMap from './components/gmap'

describe('Test the GMap Component',()=>{
    it('it should render correctly',()=>{
        const sites= [
          {
              "experiments": "MAC Season 8: Border ",
              "polygon": {
                  "type": "Polygon",
                  "coordinates": [
                    [
                        [
                            -111.97505513256,
                            33.0745586798386,
                            115
                        ],
                        [
                            -111.975038806328,
                            33.0745586755482,
                            115
                        ],
                        [
                            -111.975038816222,
                            33.074532021604,
                            115
                        ],
                        [
                            -111.975055142448,
                            33.0745320258945,
                            115
                        ],
                        [
                            -111.97505513256,
                            33.0745586798386,
                            115
                        ]
                    ]
                ]
              },
          },
          {
              "experiments": "MAC Season 8: Uniformity ",
              "polygon": {
                  "type": "Polygon",
                  "coordinates": [
                    [
                        [
                            -111.975038806328,
                            33.0745586755482,
                            115
                        ],
                        [
                            -111.975022480098,
                            33.0745586713459,
                            115
                        ],
                        [
                            -111.975022489995,
                            33.0745320173113,
                            115
                        ],
                        [
                            -111.975038816222,
                            33.074532021604,
                            115
                        ],
                        [
                            -111.975038806328,
                            33.0745586755482,
                            115
                        ]
                    ]
                ]
              }
          },
          {
              "experiments": "MAC Season 8: DP4 ",
              "polygon": {
                  "type": "Polygon",
                  "coordinates": [
                    [
                        [
                            -111.975022480098,
                            33.0745586713459,
                            115
                        ],
                        [
                            -111.975006153866,
                            33.0745586670512,
                            115
                        ],
                        [
                            -111.975006163769,
                            33.0745320130166,
                            115
                        ],
                        [
                            -111.975022489995,
                            33.0745320173113,
                            115
                        ],
                        [
                            -111.975022480098,
                            33.0745586713459,
                            115
                        ]
                    ]
                ]
              }
          }
      ];
      const features= [
        {
            "type": "Feature",
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            -111.97505513256,
                            33.0745586798386,
                            115
                        ],
                        [
                            -111.975038806328,
                            33.0745586755482,
                            115
                        ],
                        [
                            -111.975038816222,
                            33.074532021604,
                            115
                        ],
                        [
                            -111.975055142448,
                            33.0745320258945,
                            115
                        ],
                        [
                            -111.97505513256,
                            33.0745586798386,
                            115
                        ]
                    ]
                ]
            },
            "properties": {
                "Id": 1,
                "rep": 0,
                "f_range": 1,
                "f_column": 1,
                "row": "B",
                "plotid": 101,
                "treatment": 900,
                "experiment": "border",
                "variety": "tiburon in border",
                "notes": ""
            }
        },
        {
            "type": "Feature",
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            -111.975038806328,
                            33.0745586755482,
                            115
                        ],
                        [
                            -111.975022480098,
                            33.0745586713459,
                            115
                        ],
                        [
                            -111.975022489995,
                            33.0745320173113,
                            115
                        ],
                        [
                            -111.975038816222,
                            33.074532021604,
                            115
                        ],
                        [
                            -111.975038806328,
                            33.0745586755482,
                            115
                        ]
                    ]
                ]
            },
            "properties": {
                "Id": 2,
                "rep": 0,
                "f_range": 1,
                "f_column": 2,
                "row": "B",
                "plotid": 102,
                "treatment": 900,
                "experiment": "border",
                "variety": "tiburon in border",
                "notes": ""
            }
        },
        {
            "type": "Feature",
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            -111.975022480098,
                            33.0745586713459,
                            115
                        ],
                        [
                            -111.975006153866,
                            33.0745586670512,
                            115
                        ],
                        [
                            -111.975006163769,
                            33.0745320130166,
                            115
                        ],
                        [
                            -111.975022489995,
                            33.0745320173113,
                            115
                        ],
                        [
                            -111.975022480098,
                            33.0745586713459,
                            115
                        ]
                    ]
                ]
            },
            "properties": {
                "Id": 3,
                "rep": 0,
                "f_range": 1,
                "f_column": 3,
                "row": "B",
                "plotid": 103,
                "treatment": 900,
                "experiment": "border",
                "variety": "tiburon in border",
                "notes": ""
            }
        },
        {
            "type": "Feature",
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            -111.975006153866,
                            33.0745586670512,
                            115
                        ],
                        [
                            -111.974989827634,
                            33.0745586627543,
                            115
                        ],
                        [
                            -111.974989837542,
                            33.0745320087196,
                            115
                        ],
                        [
                            -111.975006163769,
                            33.0745320130166,
                            115
                        ],
                        [
                            -111.975006153866,
                            33.0745586670512,
                            115
                        ]
                    ]
                ]
            },
            "properties": {
                "Id": 4,
                "rep": 0,
                "f_range": 1,
                "f_column": 4,
                "row": "B",
                "plotid": 104,
                "treatment": 900,
                "experiment": "border",
                "variety": "tiburon in border",
                "notes": ""
            }
        }
      ];
        const bounds=[ [33.0745319570799, -111.975055142448], [33.0764637066593, -111.974793202983]];
        const colors= ["red", "blue", "green", "yellow"]
        const gmap = shallow(<GMap.WrappedComponent type="experiments"/>, {
          disableLifecycleMethods: true
        });
        gmap.setState({
            sites,
            features,
            bounds,
            colors
        });
        expect(gmap).toMatchSnapshot();
    })
})