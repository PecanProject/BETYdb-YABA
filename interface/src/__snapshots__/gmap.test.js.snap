// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the GMap Component it should render correctly 1`] = `
<div>
  <div
    className="legend"
    id="legendexperiments"
  />
  <Map
    animate={true}
    attributionControl={true}
    bounds={
      Array [
        Array [
          33.0745319570799,
          -111.975055142448,
        ],
        Array [
          33.0764637066593,
          -111.974793202983,
        ],
      ]
    }
    center={
      Array [
        33.056702,
        -112.046656,
      ]
    }
    doubleClickZoom={true}
    dragging={true}
    easeLinearity={0.35}
    maxZoom={30}
    scrollWheelZoom={true}
    touchZoom={true}
    zoom={19}
    zoomControl={true}
  >
    <ForwardRef(Leaflet(TileLayer))
      attribution="&copy <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
      url="https://{s}.tile.osm.org/{z}/{x}/{y}.png"
    />
    <ForwardRef(Leaflet(GeoJSON))
      data={
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  -111.97505513256,
                  33.0745586798386,
                  115,
                ],
                Array [
                  -111.975038806328,
                  33.0745586755482,
                  115,
                ],
                Array [
                  -111.975038816222,
                  33.074532021604,
                  115,
                ],
                Array [
                  -111.975055142448,
                  33.0745320258945,
                  115,
                ],
                Array [
                  -111.97505513256,
                  33.0745586798386,
                  115,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "properties": Object {
            "Id": 1,
            "color": "red",
            "experiment": "border",
            "f_column": 1,
            "f_range": 1,
            "notes": "",
            "plotid": 101,
            "rep": 0,
            "row": "B",
            "treatment": 900,
            "variety": "tiburon in border",
          },
          "type": "Feature",
        }
      }
      key="1"
      style={
        Object {
          "color": "#000000",
          "fillColor": "red",
          "fillOpacity": 1,
          "weight": 0.5,
        }
      }
    >
      <ForwardRef(Leaflet(Popup))>
        <p>
          Range 
          1
           Column 
          1
        </p>
      </ForwardRef(Leaflet(Popup))>
    </ForwardRef(Leaflet(GeoJSON))>
    <ForwardRef(Leaflet(GeoJSON))
      data={
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  -111.975038806328,
                  33.0745586755482,
                  115,
                ],
                Array [
                  -111.975022480098,
                  33.0745586713459,
                  115,
                ],
                Array [
                  -111.975022489995,
                  33.0745320173113,
                  115,
                ],
                Array [
                  -111.975038816222,
                  33.074532021604,
                  115,
                ],
                Array [
                  -111.975038806328,
                  33.0745586755482,
                  115,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "properties": Object {
            "Id": 2,
            "color": "blue",
            "experiment": "border",
            "f_column": 2,
            "f_range": 1,
            "notes": "",
            "plotid": 102,
            "rep": 0,
            "row": "B",
            "treatment": 900,
            "variety": "tiburon in border",
          },
          "type": "Feature",
        }
      }
      key="2"
      style={
        Object {
          "color": "#000000",
          "fillColor": "blue",
          "fillOpacity": 1,
          "weight": 0.5,
        }
      }
    >
      <ForwardRef(Leaflet(Popup))>
        <p>
          Range 
          1
           Column 
          2
        </p>
      </ForwardRef(Leaflet(Popup))>
    </ForwardRef(Leaflet(GeoJSON))>
    <ForwardRef(Leaflet(GeoJSON))
      data={
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  -111.975022480098,
                  33.0745586713459,
                  115,
                ],
                Array [
                  -111.975006153866,
                  33.0745586670512,
                  115,
                ],
                Array [
                  -111.975006163769,
                  33.0745320130166,
                  115,
                ],
                Array [
                  -111.975022489995,
                  33.0745320173113,
                  115,
                ],
                Array [
                  -111.975022480098,
                  33.0745586713459,
                  115,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "properties": Object {
            "Id": 3,
            "color": "green",
            "experiment": "border",
            "f_column": 3,
            "f_range": 1,
            "notes": "",
            "plotid": 103,
            "rep": 0,
            "row": "B",
            "treatment": 900,
            "variety": "tiburon in border",
          },
          "type": "Feature",
        }
      }
      key="3"
      style={
        Object {
          "color": "#000000",
          "fillColor": "green",
          "fillOpacity": 1,
          "weight": 0.5,
        }
      }
    >
      <ForwardRef(Leaflet(Popup))>
        <p>
          Range 
          1
           Column 
          3
        </p>
      </ForwardRef(Leaflet(Popup))>
    </ForwardRef(Leaflet(GeoJSON))>
    <ForwardRef(Leaflet(GeoJSON))
      data={
        Object {
          "geometry": Object {
            "coordinates": Array [
              Array [
                Array [
                  -111.975006153866,
                  33.0745586670512,
                  115,
                ],
                Array [
                  -111.974989827634,
                  33.0745586627543,
                  115,
                ],
                Array [
                  -111.974989837542,
                  33.0745320087196,
                  115,
                ],
                Array [
                  -111.975006163769,
                  33.0745320130166,
                  115,
                ],
                Array [
                  -111.975006153866,
                  33.0745586670512,
                  115,
                ],
              ],
            ],
            "type": "Polygon",
          },
          "properties": Object {
            "Id": 4,
            "color": "#FFFFFF",
            "experiment": "border",
            "f_column": 4,
            "f_range": 1,
            "notes": "",
            "plotid": 104,
            "rep": 0,
            "row": "B",
            "treatment": 900,
            "variety": "tiburon in border",
          },
          "type": "Feature",
        }
      }
      key="4"
      style={
        Object {
          "color": "#000000",
          "fillColor": "#FFFFFF",
          "fillOpacity": 1,
          "weight": 0.5,
        }
      }
    >
      <ForwardRef(Leaflet(Popup))>
        <p>
          Range 
          1
           Column 
          4
        </p>
      </ForwardRef(Leaflet(Popup))>
    </ForwardRef(Leaflet(GeoJSON))>
    );
  </Map>
</div>
`;
